aws/cloudfront_distribution_configuration_structure.go

```diff
@@ -43,33 +33,26 @@ func (p StringPtrSlice) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
 func expandDistributionConfig(d *schema.ResourceData) *cloudfront.DistributionConfig {
 	distributionConfig := &cloudfront.DistributionConfig{
 		CacheBehaviors:       expandCacheBehaviors(d.Get("ordered_cache_behavior").([]interface{})),
+		CallerReference:      aws.String(time.Now().Format(time.RFC3339Nano)),
+		Comment:              aws.String(d.Get("comment").(string)),
 		CustomErrorResponses: expandCustomErrorResponses(d.Get("custom_error_response").(*schema.Set)),
-		DefaultCacheBehavior: expandDefaultCacheBehavior(d.Get("default_cache_behavior").(*schema.Set).List()[0].(map[string]interface{})),
+		DefaultCacheBehavior: expandCloudFrontDefaultCacheBehavior(d.Get("default_cache_behavior").([]interface{})[0].(map[string]interface{})),
+		DefaultRootObject:    aws.String(d.Get("default_root_object").(string)),
 		Enabled:              aws.Bool(d.Get("enabled").(bool)),
 		IsIPV6Enabled:        aws.Bool(d.Get("is_ipv6_enabled").(bool)),
 		HttpVersion:          aws.String(d.Get("http_version").(string)),
 		Origins:              expandOrigins(d.Get("origin").(*schema.Set)),
 		PriceClass:           aws.String(d.Get("price_class").(string)),
+		WebACLId:             aws.String(d.Get("web_acl_id").(string)),
 	}
 
 	// This sets CallerReference if it's still pending computation (ie: new resource)
-	if v, ok := d.GetOk("caller_reference"); !ok {
-		distributionConfig.CallerReference = aws.String(time.Now().Format(time.RFC3339Nano))
-	} else {
+	if v, ok := d.GetOk("caller_reference"); ok {
 		distributionConfig.CallerReference = aws.String(v.(string))
 	}
-	if v, ok := d.GetOk("comment"); ok {
-		distributionConfig.Comment = aws.String(v.(string))
-	} else {
-		distributionConfig.Comment = aws.String("")
```

Comment:
The if/else logic here was extraneous as `d.Get("ATTR").(string)` will return the zero value (`""`), so moved above into `DistributionConfig` struct creation above.